games-fps/redeclipse/files/redeclipse

```diff
@@ -7,4 +7,4 @@
 SERVER=/usr/bin/redeclipse_server_linux
 CLIENT=/usr/bin/redeclipse_linux
 cd /usr/share/redeclipse || exit 1
-[ "$0" = "redeclipse_server" ] && exec $SERVER || exec $CLIENT
+[ "`basename $0`" = "redeclipse_server" ] && exec $SERVER || exec $CLIENT
```

Comment:
Use `$()`, always. Also you may want to use shell substitution instead (`${0##*/}`).

While at it, please also fix the other vars to use `${}` and quote them in case we introduced Prefix paths in the future.

Reply:
I've changed it to:

`[ "$(basename $0)" = "redeclipse_server" ] && exec "${SERVER}" || exec "${CLIENT}"`

After checking the bash manual, my understanding is that the usage of `$0` is the only designated way to reference the invocation name of the shell script and that, being a variable, it is considered variable substitution. Did you want me to assign its value to another variable to make it readable to those who do not know what `basename $0` or `${0##*/}` means? I'm not sure what you are asking me to do with this.

Reply:
No, I'm saying you can use `"${0##*/}"` to avoid having to call an external program (`basename`).

Reply:
(not saying you have to, alternatively just quote `"${0}"` passed to `basename`.